packages/gateway/src/manager.ts

```diff
@@ -54,6 +55,160 @@ export function createGatewayManager(options: CreateGatewayManagerOptions): Gate
+        // Refilling buckets with new values
+        gateway.prepareBuckets()
+
+        // SPREAD THIS OUT TO DIFFERENT WORKERS TO BEGIN STARTING UP
+        gateway.buckets.forEach(async (bucket, bucketId) => {
+          for (const worker of bucket.workers) {
+            for (const shardId of worker.queue) {
+              await gateway.resharding.tellWorkerToPrepare(worker.id, shardId, bucketId)
+            }
+          }
+        })
+      },
+      async tellWorkerToPrepare(workerId, shardId, bucketId) {
+        logger.warn(`[Resharding] Telling worker to prepare. Worker: ${workerId} | Shard: ${shardId} | Bucket: ${bucketId}`)
+        const shard = new Shard({
+          id: shardId,
+          connection: {
+            compress: gateway.compress,
+            intents: gateway.intents,
+            properties: gateway.properties,
+            token: gateway.token,
+            totalShards: gateway.totalShards,
+            url: gateway.url,
+            version: gateway.version,
+          },
+          // Ignore events until we are ready
+          events: {
+            async message(shard, payload) {
+              if (payload.t === 'READY') {
+                await gateway.resharding.updateGuildsShardId(
+                  (payload.d as DiscordReady).guilds.map((g) => g.id),
+                  shardId,
+                )
+              }
+            },
+          },
+          requestIdentify: async () => {
+            await gateway.identify(shardId)
+          },
+          shardIsReady: async () => {
+            logger.debug(`[Shard] Shard #${shardId} is ready`)
+            await delay(gateway.spawnShardDelay)
+            logger.debug(`[Shard] Resolving shard identify request`)
+            gateway.buckets.get(shardId % gateway.connection.sessionStartLimit.maxConcurrency)!.identifyRequests.shift()?.()
+          },
+        })
+
+        if (gateway.preferSnakeCase) {
+          shard.forwardToBot = async (payload) => {
+            options.events?.message?.(shard, payload)
+          }
+        }
+
+        gateway.resharding.shards.set(shardId, shard)
+
+        const bucket = gateway.buckets.get(shardId % gateway.connection.sessionStartLimit.maxConcurrency)
+        if (!bucket) return
+
+        return await new Promise((resolve) => {
+          // Mark that we are making an identify request so another is not made.
+          bucket.identifyRequests.push(resolve)
+          logger.debug(`[Gateway] identifying shard #(${shardId}).`)
+          // This will trigger identify and when READY is received it will resolve the above request.
+          shard?.identify().then(async () => {
+            // Tell the manager that this shard is online
+            return await gateway.resharding.shardIsPending(shard)
+          })
+        })
+      },
+      async updateGuildsShardId(guildIds, shardId) {
+        logger.warn(`[Resharding] Updating the following guild ids shard to #${shardId}: ${guildIds.join(', ')}`)
+      },
```

Comment:
Uhh is this function/method supposed to do something? Why is there only 1 `logger.warn` line?

Reply:
Do we even need this function internally? I feel like this is another function that was supposed to be overrode by the user to update their guilds shard ids _if_ they store it

Reply:
This should be changed by the user if they need, otherwise will throw a warning (will decide if it should warn or if we should just ignore this function if not overridden)